restapi/src/main/java/io/stargate/web/docsapi/service/query/condition/impl/ExistsCondition.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright The Stargate Authors
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package io.stargate.web.docsapi.service.query.condition.impl;
+
+import io.stargate.db.query.builder.BuiltCondition;
+import io.stargate.web.docsapi.service.query.condition.BaseCondition;
+import java.util.Optional;
+import org.immutables.value.Value;
+
+// TODO is this class needed at all? Can it provide a built in condition?
+//  if yes add test
+@Value.Immutable
+public abstract class ExistsCondition extends CombinedCondition<Boolean> implements BaseCondition {
+
+  /** {@inheritDoc} */
+  @Override
+  public Optional<BuiltCondition> getBuiltCondition() {
+    // TODO what's real predicate for the exists
```

Comment:
The predicate for `exists` would just be that the row trivially exists...maybe you can match against the `leaf` column? But technically as of now, for `{"a.b":{ "$exists": true}}`

A query such as `select ... from table where p0 = 'a' and p1 = 'b' and leaf = 'b'` is performed, and each row that's returned is a matching document.

I'm not sure how easy that is to translate into a BuiltCondition...

Reply:
> `select ... from table where p0 = 'a' and p1 = 'b' and leaf = 'b'`

Technically this query will also allow docs with an existing property like `a.b.c.b`, right? Is that indented?

Reply:
This is perfect, I can easily integrate that. But just to confirm you are effectively ignoring the `exists` atm with: https://github.com/stargate/stargate/blob/87998bcbf3aaec481f7cd318f32cac26d08f421b/restapi/src/main/java/io/stargate/web/docsapi/service/DocumentService.java#L1203

```

```

Reply:
> > `select ... from table where p0 = 'a' and p1 = 'b' and leaf = 'b'`
> 
> Technically this query will also allow docs with an existing property like `a.b.c.b`, right? Is that indented?

No because during the queering we are "nulling: the rest of the `p`s:
`select ... from table where p0 = 'a' and p1 = 'b' and p2 = '' and p3 = ''  and leaf = 'b'`

But I guess only one empty check is needed imo..

Reply:
In that ^ case checking `leaf` is redundant, I think. The query might be more efficient without it too, since `leaf` is not part of the PK, WDYT?